### PR TITLE
Guest can create a campaign

### PR DESCRIPTION
**Pull request by Andycutter**
## PT-link: 
https://www.pivotaltracker.com/story/show/160250436

## User story:
```
As a Guest, 
In  order to bring the Artists to my city,
I would like to create a Campaign,
```

## Tasks
- Create a page for the creation of `Campaigns`
- Implement `fields`: Title, description, location

## New Migrations
`CreateCampaigns`

## What've learned
- Repetition

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ group :development, :test do
   gem 'factory_bot_rails'
   gem 'shoulda-matchers'
   gem 'coveralls', require: false
+  gem 'rubocop-rspec'
```
Comment (tochman) [inline Gemfile]: What are you using this gem for?

File: app/controllers/campaigns_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    end
+  end
+
+private
+
+def campaign_params
+  params.require(:campaign).permit(:title, :description, :location)
+end
+
+end
```
Comment (tochman) [inline app/controllers/campaigns_controller.rb]: Please format your code

File: app/views/campaigns/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+%h1 Campaigns#index
+%p Find me in app/views/campaigns/index.html.haml
```
Comment (tochman) [inline app/views/campaigns/index.html.haml]: If you are not using this, please remove it. Sam thing with the `index` action in the controller.

File: app/views/campaigns/new.html.haml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    = form.text_area :description
+  %br/
+  .field    
+    = form.label :location
+  %br/
+  .field    
+    = form.text_field :location
+  %br/
+  .field    
+    = form.submit 'Launch Campaign'
```
Comment (tochman) [inline app/views/campaigns/new.html.haml]: There are form css in application.scss file you can use.

File: app/views/layouts/application.html.haml [hunk 5]
```diff
@@ -8,12 +8,6 @@
...
     = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
   %body
-    = render partial: 'partials/header'
-    #content
-      - flash.each do |key, value|
-        = value
-      - if current_page?(root_path)
-        = render partial: 'partials/carousell'
-      = yield
+  = yield
```
Comment (tochman) [inline app/views/layouts/application.html.haml]: I asked you to comment out the partial. Not remove it. And why did you delete the flash div?

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
   root controller: :application, action: :index
+  resources :campaigns, only: [:create, :new]
```
Comment (tochman) [inline config/routes.rb]: Nice scope or url's but why do you have an index action in the controller?

File: db/migrate/20180904135235_create_campaigns.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  def change
+    create_table :campaigns do |t|
+      t.string :title
+      t.text :description
+      t.string :location
+
+      t.timestamps
+    end
+  end
+end
```
Comment (tochman) [inline db/migrate/20180904135235_create_campaigns.rb]: This migration is also part of another PR. Why are you doing this in this sequence. Surely there must be other stories you can work on without editing the same thing twice and generate conflicts. You only have 2 weeks to work on this project.

File: features/step_definitions/assertion_steps.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+Then("I should be redirected to the Create New Campaign page") do
+    expect(current_path).to eq new_campaign_path
+end
+
+Then("there should be a Campaign titled {string} in the Database") do |expected_title|
+    expect(Campaign.last.title).to eq expected_title
```
Comment (tochman) [inline features/step_definitions/assertion_steps.rb]: I would avoid using `.last``
```ruby
campaign = Campaign.find_by(title: expected_title
expect(campaign).not_to eq nil
```
...or something

File: features/step_definitions/path_steps.rb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Given("I am on the {string} page") do |path|
+    visit root_path
```
Comment (tochman) [inline features/step_definitions/path_steps.rb]: If you don't use the variable (`path`) , don't set it.

File: app/views/campaigns/new.html.haml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  .form-group  
+    .control-label    
+      = form.label :location
+    %br/
+    .form-control    
+      = form.text_field :location
+    %br/
+  .form-group  
+    .form-submit   
+      = form.submit 'Launch Campaign'
```
Comment (ESteinman) [inline app/views/campaigns/new.html.haml]: Could you explain why this doesn't work? @tochman I feel like we probably should know this by know, but I really don't have a clue. 
For instance the styling works when we remove the `form.submit`, but obviously there's no way to submit the form then.

Comment (tochman) [reply app/views/campaigns/new.html.haml]: When you say it does not work, how is that manifested?

Comment (ESteinman) [reply app/views/campaigns/new.html.haml]: The submit-button does not exist, so there's nothing to click on.

Comment (ESteinman) [reply app/views/campaigns/new.html.haml]: screen shot 2018-09-04 at 21 21 54

Comment (ESteinman) [reply app/views/campaigns/new.html.haml]: We made progress though!

Comment (ESteinman) [reply app/views/campaigns/new.html.haml]: screen shot 2018-09-04 at 21 55 06

Now we have this!
